File: atcoder/workspace.py
import sys
input = sys.stdin.readline
from collections import deque
import heapq
from pprint import pprint
import copy

def solve(K, S, acc, C):
    # end place
    if K == 0:
        return [S]
    elif K > S[acc + C:].count('o'):
        return []
    result = []
    for i in range(acc + C, len(S)):
        if S[i] == 'o':
            temp = copy.copy(S)
            temp[i] = 'c'
            result += solve(K - 1, temp, i, C)
    return result


def main():
    N, K, C = map(int, input().rstrip().split())
    S = input()
    S = S.replace('x', ' ').rstrip().replace(' ', 'x')
    if len(S.replace('x', '')) < K:
        return
    S = list(S)
    first = 0 - C
    res = solve(K, ['x'] + S, first, C)
    # pprint(res)
    result = [0 for _ in range(N + 1)]
    for r in res:
        for i, x in enumerate(r):
            if x == 'c':
                result[i] += 1
    for idx, r in enumerate(result):
        if r == len(res):
            print(idx)


if __name__ == "__main__":
    main()

File: atcoder/workspace.cpp
#include <bits/stdc++.h>
#include <iostream>
using namespace std;
#define rep(i, n) for (int i = 0; i < (int)(n); i++)

int count(string s, char c) {
  auto res = 0;
  for(auto item: s) {
    if (item == c) res++;
  }
  return res;
}

vector<string> solve(int K, string S, int acc, int C) {
  vector<string> result;
  if (K == 0) {
    result.push_back(S);
  } else if (acc + C > S.size()) {

  } else if (K > count(string(S.begin() + acc + C, S.end()), 'o')) {
  } else {
    for (int i = acc + C; i < S.size(); ++i) {
      if(S.at(i) == 'o') {
        auto temp = string(S.begin(), S.end());
        temp[i] = 'c';
        auto _result = solve(K - 1, temp, i + 1, C);
        copy(_result.begin(), _result.end(), back_inserter(result));
      }
    }
  }
  return result;
}


int main(void) {
  int N, K, C;
  cin >> N >> K >> C;
  vector<string> results;
  string S;
  cin >> S;
  auto result = solve(K, S, 0 - C, C);
  vector<int> temp = vector<int>(N, 0);
  rep(i, N) {
    for(auto res: result) {
      if (res[i] == 'c') {
        temp.at(i) += 1;
      }
    }
  }

  rep(i, temp.size()){
    if (temp[i] == result.size())
      cout << i + 1 << endl;
  }
  

  return 0;
}
